src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -48,9 +48,13 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(MESSAGE_MARK_MARKED_TASK);
         }
 
-        Task markedTask = new Task(taskToMark.getDescription(), status.updateStatus());
-        model.setTask(taskToMark, markedTask);
-
+        if (taskToMark.getDeadline() == null) {
+            Task markedTask = new Task(taskToMark.getDescription(), status.updateStatus());
+            model.setTask(taskToMark, markedTask);
+        } else {
+            Task markedTask = new Task(taskToMark.getDescription(), status.updateStatus(), taskToMark.getDeadline());
+            model.setTask(taskToMark, markedTask);
```

Comment:
Should we have a `NullDeadline` object that extends from `Deadline` to represent no deadline assigned to the Task, rather than just using `null`?